### PR TITLE
Implement Metis SACLOS guidance

### PR DESCRIPTION
**Pull request by TheCandianVendingMachine**
**When merged this pull request will:**
- Implement SACLOS guidance for default Vorona missile
- Move SACLOS and Wire guidance to Missile Guidance framework

### CODE REVIEW

File: addons/missileguidance/script_component.hpp [hunk 1]
```diff
@@ -16,3 +16,14 @@
 #endif
 
 #include "\z\ace\addons\main\script_macros.hpp"
+
+#define RANDOM_VECTOR_3D (call {\
+    private _z = random 2 - 1;\
+    private _r = sqrt (1 - _z^2);\
+    private _theta = random 360;\
+    [_r * cos _theta, _r * sin _theta, _z]\
+})
```
Comment (commy2) [inline addons/missileguidance/script_component.hpp]: Archimedes!

File: addons/missileguidance/functions/fnc_SACLOS_onFired.sqf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * Public: No
+ */
+params ["_firedEH", "", "", "", "_stateParams"];
+_firedEH params ["_shooter","_weapon","","","","","_projectile"];
+_stateParams params ["", "_seekerStateParams"];
+
+private _config = ([_projectile] call CBA_fnc_getObjectConfig) >> "ace_missileguidance";
+private _distanceAheadOfMissile = [_config >> "missileLeadDistance", "NUMBER", DEFAULT_LEAD_DISTANCE] call CBA_fnc_getConfigEntry;
+
+if (_shooter isKindOf "Plane") then { WARNING("SACLOS fired from planes unsupported"); };
```
Comment (TheMagnetar) [inline addons/missileguidance/functions/fnc_SACLOS_onFired.sqf]: if unsupported, should then the function exit?

File: addons/missileguidance/script_component.hpp (hunk 1 above)
Comment (TheCandianVendingMachine) [reply addons/missileguidance/script_component.hpp]: What do you mean?

Comment (commy2) [reply addons/missileguidance/script_component.hpp]: This random distribution function is based on the fact that a sphere and a cylinder with height = radius have the same surface area. Something that Archimedes figured out.

Comment (TheCandianVendingMachine) [reply addons/missileguidance/script_component.hpp]: Well don't give me the credit for coming up with it, give it to yourself :P https://github.com/acemod/ACE3/pull/6708#discussion_r237926302

Comment (commy2) [reply addons/missileguidance/script_component.hpp]: Hmm, maybe I should create it as a CBA function. It seems to be used in multiple places now.

Comment (SilentSpike) [reply addons/missileguidance/script_component.hpp]: @commy2 Create it as a function, but generalise it to `n` dimensions ;)

**Pull request merged**